eng/common/TestResources/New-TestResources.ps1

```diff
@@ -351,6 +354,15 @@ try {
     # Enumerate test resources to deploy. Fail if none found.
     $repositoryRoot = "$PSScriptRoot/../../.." | Resolve-Path
     $root = [System.IO.Path]::Combine($repositoryRoot, "sdk", $ServiceDirectory) | Resolve-Path
+    if ($TestResourcesDirectory) {
+        $root = $TestResourcesDirectory | Resolve-Path
+        # Add an explicit check below in case ErrorActionPreference is overridden and Resolve-Path doesn't stop execution
+        if (!$root) {
+            Write-Error "TestResourcesDirectory '$TestResourcesDirectory' does not exist."
+            exit 1
```

Comment:
Tip: you could just: `throw "TestResourcesDirectory '$TestResourcesDirectory' does not exist."`. You get 2 for the price of 1.